### ISSUE TITLE
sql: move temp tables out of experimental

### ISSUE THREAD

**Issue by otan** [cockroachdb/cockroach#46260]
Currently, temp tables (/views/sequences) are experimental. This issues is be a "master issue" of things remaining to do before moving it off experimental.

Things to do:
 - [ ] performance benchmarks (https://github.com/cockroachdb/cockroach/issues/46213)
 - [ ] move descriptors from gossiping all descriptors to only gossiping deltas

**Comment by andreimatei** [comment]
I think there were more problems than listed above with the temp tables. I think we'd want an in-memory implementation for them by default, and some smarts about the locality in which they're created.
There was a lot of discussion on the RFC; have you found those threads?

**Comment by otan** [comment]
Nope, this (so far) isn't a definitive list (I just wanted to create the issue). Feel free to edit!

**Comment by jpollard-cs** [comment]
Hello just curious if there's any update on this? Appreciating that this is likely more complex in the Cockroach case for various reasons I can see this being a potential question for our internal customers deciding between Cockroach and Aurora  given Aurora uses local storage for temporary tables which can significantly improve read performance for some workloads.
